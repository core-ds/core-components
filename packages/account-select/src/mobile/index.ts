export { AccountSelectMobile } from './Component.mobile';
