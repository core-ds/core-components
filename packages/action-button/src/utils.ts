import { getDataTestId } from '@alfalab/core-components-shared';

export function getActionButtonTestIds(dataTestId: string) {
    return {
        actionButton: dataTestId,
        spinner: getDataTestId(dataTestId, 'loader'),
    };
}
