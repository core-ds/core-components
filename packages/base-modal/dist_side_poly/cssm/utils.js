'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreComponentsGlobalStore = require('@alfalab/core-components-global-store/cssm');
var coreComponentsShared = require('@alfalab/core-components-shared/cssm');

function isScrolledToTop(target) {
    return target.scrollTop <= 0;
}
function isScrolledToBottom(target) {
    return target.scrollHeight - target.offsetHeight <= target.scrollTop;
}
function hasScrollbar(target) {
    return target.scrollHeight > target.clientHeight;
}
var isOverflowing = function (container) {
    if (document.body === container) {
        return window.innerWidth > document.documentElement.clientWidth;
    }
    return container.scrollHeight > container.clientHeight;
};
var getPaddingRight = function (node) {
    return parseInt(window.getComputedStyle(node).paddingRight, 10) || 0;
};
var restoreContainerStyles = function (container) {
    var modalRestoreStyles = coreComponentsGlobalStore.getModalStore().getRestoreStyles();
    var index = modalRestoreStyles.findIndex(function (s) { return s.container === container; });
    var existingStyles = modalRestoreStyles[index];
    if (!existingStyles)
        return;
    existingStyles.modals -= 1;
    if (existingStyles.modals <= 0) {
        modalRestoreStyles.splice(index, 1);
        existingStyles.styles.forEach(function (_a) {
            var value = _a.value, el = _a.el, key = _a.key;
            if (value) {
                el.style.setProperty(key, value);
            }
            else {
                el.style.removeProperty(key);
            }
        });
    }
};
var handleContainer = function (container, shouldIOSLock) {
    if (shouldIOSLock === void 0) { shouldIOSLock = false; }
    if (!container)
        return;
    var modalRestoreStyles = coreComponentsGlobalStore.getModalStore().getRestoreStyles();
    var existingStyles = modalRestoreStyles.find(function (s) { return s.container === container; });
    if (existingStyles) {
        existingStyles.modals += 1;
        return;
    }
    var containerStyles = [];
    if (isOverflowing(container)) {
        // Вычисляет размер до применения `overflow hidden` для избежания скачков
        var scrollbarSize = coreComponentsShared.browser.getScrollbarSize();
        containerStyles.push({
            value: container.style.paddingRight,
            key: 'padding-right',
            el: container,
        });
        // Вычисляем стили, чтобы получить реальный `padding` c шириной сколлбара
        // eslint-disable-next-line no-param-reassign
        container.style.paddingRight = "".concat(getPaddingRight(container) + scrollbarSize, "px");
    }
    var parent = container.parentElement;
    var scrollContainer = 
    // TODO: заменить на optional chaining
    parent &&
        parent.nodeName === 'HTML' &&
        window.getComputedStyle(parent).overflowY === 'scroll'
        ? parent
        : container;
    // Блокируем скролл даже если отсутствует скроллбар
    if (scrollContainer.style.overflow !== 'hidden') {
        containerStyles.push({
            value: scrollContainer.style.overflow,
            key: 'overflow',
            el: scrollContainer,
        });
    }
    if (!shouldIOSLock) {
        scrollContainer.style.overflow = 'hidden';
    }
    modalRestoreStyles.push({
        container: container,
        modals: 1,
        styles: containerStyles,
    });
};

exports.handleContainer = handleContainer;
exports.hasScrollbar = hasScrollbar;
exports.isScrolledToBottom = isScrolledToBottom;
exports.isScrolledToTop = isScrolledToTop;
exports.restoreContainerStyles = restoreContainerStyles;
