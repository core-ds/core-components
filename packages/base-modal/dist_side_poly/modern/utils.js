import { getModalStore } from '@alfalab/core-components-global-store/modern';
import { browser } from '@alfalab/core-components-shared/modern';

function isScrolledToTop(target) {
    return target.scrollTop <= 0;
}
function isScrolledToBottom(target) {
    return target.scrollHeight - target.offsetHeight <= target.scrollTop;
}
function hasScrollbar(target) {
    return target.scrollHeight > target.clientHeight;
}
const isOverflowing = (container) => {
    if (document.body === container) {
        return window.innerWidth > document.documentElement.clientWidth;
    }
    return container.scrollHeight > container.clientHeight;
};
const getPaddingRight = (node) => parseInt(window.getComputedStyle(node).paddingRight, 10) || 0;
const restoreContainerStyles = (container) => {
    const modalRestoreStyles = getModalStore().getRestoreStyles();
    const index = modalRestoreStyles.findIndex((s) => s.container === container);
    const existingStyles = modalRestoreStyles[index];
    if (!existingStyles)
        return;
    existingStyles.modals -= 1;
    if (existingStyles.modals <= 0) {
        modalRestoreStyles.splice(index, 1);
        existingStyles.styles.forEach(({ value, el, key }) => {
            if (value) {
                el.style.setProperty(key, value);
            }
            else {
                el.style.removeProperty(key);
            }
        });
    }
};
const handleContainer = (container, shouldIOSLock = false) => {
    if (!container)
        return;
    const modalRestoreStyles = getModalStore().getRestoreStyles();
    const existingStyles = modalRestoreStyles.find((s) => s.container === container);
    if (existingStyles) {
        existingStyles.modals += 1;
        return;
    }
    const containerStyles = [];
    if (isOverflowing(container)) {
        // Вычисляет размер до применения `overflow hidden` для избежания скачков
        const scrollbarSize = browser.getScrollbarSize();
        containerStyles.push({
            value: container.style.paddingRight,
            key: 'padding-right',
            el: container,
        });
        // Вычисляем стили, чтобы получить реальный `padding` c шириной сколлбара
        // eslint-disable-next-line no-param-reassign
        container.style.paddingRight = `${getPaddingRight(container) + scrollbarSize}px`;
    }
    const parent = container.parentElement;
    const scrollContainer = 
    // TODO: заменить на optional chaining
    parent &&
        parent.nodeName === 'HTML' &&
        window.getComputedStyle(parent).overflowY === 'scroll'
        ? parent
        : container;
    // Блокируем скролл даже если отсутствует скроллбар
    if (scrollContainer.style.overflow !== 'hidden') {
        containerStyles.push({
            value: scrollContainer.style.overflow,
            key: 'overflow',
            el: scrollContainer,
        });
    }
    if (!shouldIOSLock) {
        scrollContainer.style.overflow = 'hidden';
    }
    modalRestoreStyles.push({
        container,
        modals: 1,
        styles: containerStyles,
    });
};

export { handleContainer, hasScrollbar, isScrolledToBottom, isScrolledToTop, restoreContainerStyles };
