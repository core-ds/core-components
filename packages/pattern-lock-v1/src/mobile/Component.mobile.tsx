import React, { forwardRef } from 'react';
import type { TPatternLockInstance } from 'react-canvas-pattern-lock';

import { BasePatternLock } from '../components/base-pattern-lock';
import { CommonPatternLockProps } from '../typings';

import styles from './mobile.module.css';

export const PatternLockMobileV1 = forwardRef<TPatternLockInstance, CommonPatternLockProps>(
    (restProps, ref) => <BasePatternLock {...restProps} ref={ref} styles={styles} />,
);
