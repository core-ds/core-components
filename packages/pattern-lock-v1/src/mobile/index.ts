export { PatternLockMobileV1 } from './Component.mobile';
export type { CommonPatternLockProps as PatternLockMobileProps } from '../typings';
