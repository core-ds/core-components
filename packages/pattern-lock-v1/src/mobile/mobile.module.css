@import '../../../themes/src/default.css';
@import '../vars.css';

.component {
    max-width: var(--pattern-lock-v1-max-width);
    margin: var(--gap-xs) 0 var(--gap-xl) 0;
}

.forgotBtn {
    margin: var(--gap-xl) auto 0;
}

@media screen and (min-width: 360px) {
    .forgotBtn {
        margin-top: var(--gap-2xl);
        min-height: var(--size-s-height);
    }

    .component {
        margin-bottom: var(--gap-s);
    }
}

@media screen and (min-width: 390px) {
    .forgotBtn {
        min-height: var(--size-m-height);
    }

    .component {
        margin-bottom: var(--gap-l);
    }
}
