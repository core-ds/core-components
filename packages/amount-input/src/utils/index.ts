/**
 * https://github.com/core-ds/utils
 * TODO: нужно перенести утилиты в этот проект, когда он будет готов
 */

import { CurrencyCodes } from '@alfalab/data';
import { formatAmount } from '@alfalab/utils';

/**
 * Форматирует введенное значение
 * @param enteredValue Значение введенное в инпут
 * @param currency валюта
 * @param minority количество минорных единиц
 */
export function getFormattedValue(enteredValue: string, currency: CurrencyCodes, minority: number) {
    if (!enteredValue || enteredValue === '-') {
        return enteredValue;
    }

    // eslint-disable-next-line prefer-const
    let [head, tail] = enteredValue.split(',');

    // При вводе "-," указываем, что имеется в виду "-0,"
    if (head === '-') {
        head = '-0';
    }

    let { majorPart } = formatAmount({
        value: Number(head) * minority,
        currency,
        minority,
        negativeSymbol: 'hyphen-minus',
    });

    // Так как -0 === 0, formatAmount возвращает положительное значение. Исправляем это здесь
    if (/^-(,|0),?/.test(enteredValue) && majorPart === '0') {
        majorPart = `-${majorPart}`;
    }

    if (!tail && enteredValue.includes(',')) {
        return majorPart.concat(',');
    }

    if (tail) {
        return majorPart.concat(',', tail.slice(0, minority.toString().length - 1));
    }

    return majorPart;
}

export function getAmountValueFromStr(str: string, minority: number) {
    if (str === '' || str === '-') {
        return null;
    }

    return Math.round(Number(str.replace(',', '.').replace(/[^0-9.-]/g, '')) * minority);
}
