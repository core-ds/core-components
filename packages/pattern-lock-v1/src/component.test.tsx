import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { PatternLockMobileV1 as PatternLockV1 } from './mobile';
import { getPatternLockTestIds, getSizes } from './utils';

jest.mock('react-canvas-pattern-lock', () => {
    return {
        ReactCanvasPatternLock: jest.fn((props) => null),
    };
});

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => {
        return {
            matches: query === '(min-width: 390px)',
            media: query,
            addListener: jest.fn(),
            removeListener: jest.fn(),
        };
    }),
});

const mutationObserverMock = jest.fn<any, any>(function MutationObserver(this: MutationObserver) {
    this.observe = jest.fn();
    this.disconnect = jest.fn();
});
global.MutationObserver = mutationObserverMock;

describe('PatternLockV1 test', () => {
    describe('Snapshot tests', () => {
        it('should match snapshot', () => {
            const { container } = render(<PatternLockV1 />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with error prop', () => {
            const { container } = render(<PatternLockV1 error='Error message' />);

            expect(container).toMatchSnapshot();
        });
    });

    describe('attrubute tests', () => {
        it('should set `data-test-id` attribute', () => {
            const dti = 'pattern-lock-dti';
            const { getByTestId } = render(
                <PatternLockV1
                    message='message'
                    onForgotBtnClick={jest.fn()}
                    forgotCodeBtnText='кнопка'
                    showForgotCodeBtn={true}
                    dataTestId={dti}
                />,
            );

            const testIds = getPatternLockTestIds(dti);

            expect(getByTestId(testIds.patternLock)).toBeInTheDocument();
            expect(getByTestId(testIds.forgotCodeBtn)).toBeInTheDocument();
            expect(getByTestId(testIds.message)).toBeInTheDocument();

            const { getByTestId: getByTestIdError } = render(
                <PatternLockV1 error='error' dataTestId={dti} />,
            );

            expect(getByTestIdError(testIds.error)).toBeInTheDocument();
        });

        it('should set `className` class', () => {
            const className = 'test-class';
            const { container } = render(<PatternLockV1 className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should shown message', () => {
            const message = 'message';
            const { getByText } = render(<PatternLockV1 message={message} />);

            const messageElement = getByText(message);

            expect(messageElement).toBeInTheDocument();
        });

        it('should shown error message', () => {
            const errorMessage = 'Error message';
            const { getByText } = render(<PatternLockV1 error={errorMessage} />);

            const errorElement = getByText(errorMessage);

            expect(errorElement).toBeInTheDocument();
        });

        it('should call onForgotBtnClick', () => {
            const testId = 'pattern-lock';
            const onForgotBtnClick = jest.fn();
            const { getByTestId } = render(
                <PatternLockV1
                    showForgotCodeBtn={true}
                    onForgotBtnClick={onForgotBtnClick}
                    dataTestId={testId}
                />,
            );

            const buttonEl = getByTestId(`${testId}-forgot-code-btn`);

            fireEvent.click(buttonEl);

            expect(buttonEl).toBeInTheDocument();
            expect(onForgotBtnClick).toBeCalledTimes(1);
        });
    });

    describe('utils test', function () {
        it('should return "l" size', function () {
            expect(getSizes().width).toBe(322);
        });
    });

    it('should be created mutation observer instance', () => {
        render(<PatternLockV1 observeTokens={true} />);

        const [observerInstance] = mutationObserverMock.mock.instances;
        expect(observerInstance.observe).toHaveBeenCalledTimes(1);

        expect(mutationObserverMock.mock.instances).toHaveLength(1);
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<PatternLockV1 />);

        expect(unmount).not.toThrowError();
    });
});
