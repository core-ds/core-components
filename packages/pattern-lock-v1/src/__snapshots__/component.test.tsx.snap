// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatternLockV1 test Snapshot tests should match snapshot 1`] = `
<div>
  <div
    class="component component"
  >
    <div
      class="message"
    />
    <div
      class="forgotBtn forgotBtn hiddenBtn"
    />
  </div>
</div>
`;

exports[`PatternLockV1 test Snapshot tests should match snapshot with error prop 1`] = `
<div>
  <div
    class="component component"
  >
    <div
      class="message error"
    >
      Error message
    </div>
    <div
      class="forgotBtn forgotBtn hiddenBtn"
    />
  </div>
</div>
`;
